kolibri/core/assets/src/heartbeat.js

```diff
@@ -39,17 +40,26 @@ class HeartBeat {
   setInactive() {
     this.active = false;
   }
+  wait() {
+    this.timerId = setTimeout(this.beat, this.delay);
+    return this.timerId;
+  }
   beat() {
     if (this.active) {
       logging.debug('There was activity, polling session endpoint!');
       this.kolibri.resources.SessionResource.getModel('current').fetch({}, true).catch((error) => {
         logging.error('Periodic server polling failed, with error: ', error);
       });
       this.setActivityListeners();
+    } else {
+      logging.debug('No user activity, polling keepalive endpoint to keep session active');
+      this.kolibri.resources.KeepAliveResource.getModel('now').fetch({}, true).catch((error) => {
+        logging.error('Periodic server polling failed with error: ', error);
+      });
```

Comment:
You need to add another `setTimeout` here, otherwise use `setInterval` ? Or should how do we ensure that the session is kept alive through for instance watching a 15 min video without doing anything else?